### PR TITLE
refactor: showing options < 5 if search text is present

### PR DESCRIPTION
**Pull request by anandtiwary**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/multi-select/multi-select.component.ts [hunk 1]
```diff
@@ -209,9 +213,11 @@ export class MultiSelectComponent<V> implements ControlValueAccessor, AfterConte
 
   public ngAfterContentInit(): void {
     this.allOptions$ = this.allOptionsList !== undefined ? queryListAndChanges$(this.allOptionsList) : EMPTY;
-    this.filteredOptions$ = combineLatest([this.allOptions$, this.searchSubject]).pipe(
+    this.filteredOptions$ = combineLatest([this.allOptions$, this.caseInsensitiveSearchSubject]).pipe(
       map(([options, searchText]) =>
-        options.filter(option => option.label.toLowerCase().includes(searchText.toLowerCase()))
+        isEmpty(searchText)
+          ? options.map(option => option)
```
Comment (arjunlalb) [inline projects/components/src/multi-select/multi-select.component.ts:219]: we don't need the map. just `options` will do

File: projects/observability/src/shared/graphql/request/handlers/explore/explore-query.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export interface GraphQlExploreResult {
   [key: string]: GraphQlExploreResultValue;
 }
 
-type GraphQlExploreValueType = number | string | boolean | undefined;
+type GraphQlExploreValueType = number | string | string[] | boolean | undefined;
```
Comment (arjunlalb) [inline projects/observability/src/shared/graphql/request/handlers/explore/explore-query.ts:38]: is this intentional?

File: projects/components/src/multi-select/multi-select.component.ts [hunk 3]
```diff
@@ -197,7 +195,13 @@ export class MultiSelectComponent<V> implements ControlValueAccessor, AfterConte
   public allOptions$!: Observable<QueryList<SelectOptionComponent<V>>>;
 
   public filteredOptions$!: Observable<SelectOptionComponent<V>[]>;
-  private readonly searchSubject: Subject<string> = new BehaviorSubject('');
+  private readonly caseInsensitiveSearchSubject: BehaviorSubject<string> = new BehaviorSubject('');
+
+  public isSearchTextPresent$: Observable<boolean> = this.searchValueChange.pipe(
+    map(searchText => !isEmpty(searchText)),
+    startWith(false),
+    shareReplay(1)
```
Comment (arjunlalb) [inline projects/components/src/multi-select/multi-select.component.ts:203]: do we need the `shareReplay` here?

File: projects/observability/src/shared/graphql/request/handlers/explore/explore-query.ts (hunk 2 above)
Comment (anandtiwary) [reply projects/observability/src/shared/graphql/request/handlers/explore/explore-query.ts:38]: yes. dataTypeIds and labels are all string array fields. The values are returned as string[]

File: projects/components/src/multi-select/multi-select.component.ts (hunk 3 above)
Comment (anandtiwary) [reply projects/components/src/multi-select/multi-select.component.ts:203]: I think we do. Spent a lot of time fixing this test.

File: projects/components/src/multi-select/multi-select.component.ts (hunk 1 above)
Comment (anandtiwary) [reply projects/components/src/multi-select/multi-select.component.ts:219]: no. Options is of type QueryList. Some weird angular thing. When we iterate over it then we get the underlying type. Map is required.

**Review by arjunlalb — APPROVED** [review]

**Pull request merged**